src/Horse.Core.pas

```diff
@@ -5,6 +5,80 @@ interface
+    function Patch(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Patch(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Head(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Head(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Head(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Head(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Post(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Post(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Post(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Post(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Delete(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Delete(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Delete(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Delete(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+  end;
+
+  THorseCoreRoute = class(TInterfacedObject, IHorseCoreRoute)
+  private
+    FPath:string;
+    FHorseCore:THorseCore;
+
+  public
+    constructor Create(APath:string; AHorseCore: THorseCore);
+
+    function Get(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Get(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Get(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Get(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Put(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Put(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Put(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Put(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Patch(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Patch(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Patch(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Patch(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Head(ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Head(AMiddleware, ACallback: THorseCallback):IHorseCoreRoute; overload;
+    function Head(ACallbacks: array of THorseCallback):IHorseCoreRoute; overload;
+    function Head(ACallbacks: array of THorseCallback; ACallback: THorseCallback):IHorseCoreRoute; overload;
+
+    function Post(ACallback: THorseCallback):IHorseCoreRoute; overload;
```

Comment:
In all functions, place a space after the colon of the function:

`function Post(ACallback: THorseCallback):IHorseCoreRoute; overload;`

to

`function Post(ACallback: THorseCallback): IHorseCoreRoute; overload;`